openfisca_core/legislationsxml.py

```diff
@@ -3,926 +3,198 @@
-            return value, unicode(error)
-        xml_legislation = legislation_tree.getroot()
-        return xml_legislation, None
-
-    return xml_legislation_file_path_to_xml
-
-
-def make_xml_legislation_info_list_to_xml_elements_and_paths(with_source_file_infos):
-    return conv.uniform_sequence(
-        conv.struct([
-            make_xml_legislation_file_path_to_xml(with_source_file_infos),
-            conv.pipe(
-                conv.test_isinstance((list, tuple)),
-                conv.uniform_sequence(conv.test_isinstance(basestring)),
-                ),
-            ]),
-        )
-
-
-def xml_legislation_to_json(xml_element, state = None):
-    if xml_element is None:
-        return None, None
-    json_key, json_element = translate_xml_element_to_json_item(xml_element)
-    if json_key != 'NODE':
-        if state is None:
-            state = conv.default_state
-        return json_element, state._(u'Invalid root element in XML: "{}" instead of "NODE"').format(xml_element.tag)
-    return json_element, None
-
-
-# Level 2 converters
-
-def make_xml_legislation_info_list_to_xml_element(with_source_file_infos):
-    return conv.pipe(
-        make_xml_legislation_info_list_to_xml_elements_and_paths(with_source_file_infos),
-        merge_xml_elements_and_paths_into_first,
-        )
-
-
-# Used by taxbenefitsystems.MultipleXmlBasedTaxBenefitSystem
-
-def make_xml_legislation_info_list_to_json(with_source_file_infos):
-    return conv.pipe(
-        make_xml_legislation_info_list_to_xml_element(with_source_file_infos),
-        xml_legislation_to_json,
-        validate_legislation_xml_json,
-        conv.function(lambda value: transform_node_xml_json_to_json(value)[1]),
-        )
+
+def transform(xml_trees):
+    name_list = []
+    json_list = []
+    for tree in xml_trees:
+        root = tree.getroot()
+        name, json_data = transform_recursive(root)
+        name_list.append(name)
+        json_list.append(json_data)
+    return name_list, json_list
+
+
+def merge(name_list, json_list, path_list):
+    # The first json tree is special
+    merged_json = json_list[0]
+    assert not path_list[0]
+
+    for name, json_tree, path in zip(name_list, json_list, path_list)[1:]:
+        pointer = merged_json
+        for key in path:
+            if key in pointer['children']:
+                pointer = pointer['children'][key]
+            else:
+                pointer['children'][key] = {
+                    '@type': 'Node',
+                    'children': {},
+                    }
+
+        assert name not in pointer['children'], '{} is defined twice'.format('.'.join(path) + '.' + 'name')
+        pointer['children'][name] = json_tree
+
+    return merged_json
+
+
+def load_legislation(legislation_xml_info_list):
+    xmlschema = load_xml_schema()
+
+    xml_trees = parse_and_validation_xml(xmlschema, legislation_xml_info_list)
+
+    name_list, json_list = transform(xml_trees)
```

Comment:
`transform` is a bit vague. What kind of transformation is done by this function ?
Does it mostly convert XML to JSON ?